### DISCUSSION TITLE
Add the ability to change default notation?

### DISCUSSION THREAD

**Discussion by Wonton36** [godotengine/godot-proposals#5551] [Scripting]
I'm somewhat new to programming so I don't know how feasible this would be, but it would be really nice if there was an option you could select that let you change the default notation to various things like

* (of course) snake_case
* camelCase
* PascelCase
* etc.

You would be able to set the default notation for functions, nodes, etc. based on your own preferences

Again, I'm new but I don't really see a reason in enforcing a particular way of doing notation in Godot, unless there are technical limitations of course.

What do you guys think? Would this be a nice feature or no?

**Comment by Calinou** [comment]
This is already provided in the Editor Settings and Project Settings to an extent. This will be done more consistently in 4.0, but GDScript's core API will remain snake_case no matter what (and PascalCase in C#).
